js/scripts/util.js

```diff
@@ -133,4 +140,86 @@ const checkBlockCollisions = (entity) => {
     entity.boundingBox = new BoundingBox(entity.pos, entity.constructor.SCALED_SIZE);
 
     return collisions;
+};
+
+// The following is necessary because we must change the listeners for different modes (right now, gameplay and dialog).
+/** Contains all functions called as event handlers. */
+const EVENT_HANDLERS = {
+    gameplayMouseDown: (e) => {
+        GAME.user.firing = false;
+        GAME.user.aiming = true;
+    },
+    gameplayMouseUp: (e) => {
+        GAME.user.aiming = false;
+        GAME.user.firing = true;
+    },
+    gameplayMouseMove: (e) => {
+        const rect = CANVAS.getBoundingClientRect();
+        const scaleX = CANVAS.width / rect.width;
+        const scaleY = CANVAS.height / rect.height;
+        GAME.mousePos = new Vector((e.clientX - rect.left) * scaleX, (e.clientY - rect.top) * scaleY);
+    },
+    gameplayKeyDown: (e) => {
```

Comment:
Similar to what we have in some GameEngine code we could likely put each of our keys into an array and loop over it in a for loop to clean it up. Not necessary but might be nice to clean up more. Also why are we doing this in Util? Can you explain that?

Reply:
Listeners are necessary in util because we must define the event handlers separately, because they are necessary for the "removeEventHandler" function call. Simply copy pasting or similar techniques do not work, because it would not actually be the same function (same place in memory), just a function which does the same thing.

Reply:
Okay now it makes sense why these functions are in util. I see what you mean about the cleaning up of the code, I suppose if we were to do that it would not actually simplify it by using a loop.